tests/unit/customizations/s3/test_results.py

```diff
@@ -0,0 +1,204 @@
+        return self._get_transfer_future(result=result)
+
+    def get_failed_transfer_future(self, exception):
+        return self._get_transfer_future(exception=exception)
+
+    def _get_transfer_future(self, result=None, exception=None):
+        call_args = self._get_transfer_future_call_args()
+        meta = FakeTransferFutureMeta(size=self.size, call_args=call_args)
+        return FakeTransferFuture(
+            result=result, exception=exception, meta=meta)
+
+    def _get_transfer_future_call_args(self):
+        return FakeTransferFutureCallArgs(
+            fileobj=self.filename, key=self.key, bucket=self.bucket)
+
+    def get_queued_result(self):
+        return self.result_queue.get(block=False)
+
+    def assert_result_queue_is_empty(self):
+        self.assertTrue(self.result_queue.empty())
+
+    def assert_expected_transfer_type(self, result):
+        self.assertEqual(result.transfer_type, self.transfer_type)
+
+    def assert_expected_src_and_dest(self, result):
+        self.assertEqual(result.src, self.src)
+        self.assertEqual(result.dest, self.dest)
+
+    def assert_expected_total_transfer_size(self, result):
+        self.assertEqual(result.total_transfer_size, self.size)
+
+    def assert_correct_queued_result(self, result):
+        self.assertIsInstance(result, QueuedResult)
+        self.assert_expected_transfer_type(result)
+        self.assert_expected_src_and_dest(result)
+        self.assert_expected_total_transfer_size(result)
+
+    def assert_correct_progress_result(self, result, ref_bytes_transferred):
+        self.assertIsInstance(result, ProgressResult)
+        self.assert_expected_transfer_type(result)
+        self.assert_expected_src_and_dest(result)
+        self.assertEqual(result.bytes_transferred, ref_bytes_transferred)
+        self.assert_expected_total_transfer_size(result)
+
+    def assert_correct_success_result(self, result):
+        self.assertIsInstance(result, SuccessResult)
+        self.assert_expected_transfer_type(result)
+        self.assert_expected_src_and_dest(result)
+
+    def assert_correct_exception_result(self, result, ref_exception):
+        self.assertIsInstance(result, FailureResult)
+        self.assert_expected_transfer_type(result)
+        self.assert_expected_src_and_dest(result)
+        self.assertEqual(result.exception, ref_exception)
+
+
+class TestUploadResultSubscriber(BaseResultSubscriberTest):
+    def setUp(self):
+        super(TestUploadResultSubscriber, self).setUp()
+        self.src = relative_path(self.filename)
+        self.dest = 's3://' + self.bucket + '/' + self.key
+        self.transfer_type = 'upload'
+        self.result_subscriber = UploadResultSubscriber(self.result_queue)
+
+    def test_on_queued(self):
+        self.result_subscriber.on_queued(self.future)
+        result = self.get_queued_result()
+        self.assert_result_queue_is_empty()
+        self.assert_correct_queued_result(result)
```

Comment:
I think we've chatted about this before, but I prefer the test logic is actually embedded in the test.  What specifically about on queued are you testing?  That should be apparent in the test.  Right now I have no idea what constitutes as "assert_correct_queued_result" should even mean.

These are named tuples, why not just compare them to the expected result?